File: learnPython/num226.py
class Solution:
    def invertTree(self, root):
        """
        :type root: TreeNode
        :rtype: TreeNode
        """
        if root == None:
            return root
        tmp = root.left
        root.left =root.right
        root.right = tmp
        self.invertTree(root.left)
        self.invertTree(root.right)
        return root

File: maven-leetcode/src/main/java/leetcode/Num226.java
package leetcode;

public class Num226 {
    class Solution {
        public TreeNode invertTree(TreeNode root) {
            /**exchange each tree's left child and right child recursively*/
            if( root == null){return root;}
            TreeNode left = root.left;
            root.left = root.right;
            root.right = left;
            invertTree(root.left);
            invertTree(root.right);
            return root;
        }
    }
}
